### PR TITLE
Remove `CoreStringNames` and `SceneStringNames`

### PR DESCRIPTION
**Pull request by AaronRecord**
See https://chat.godotengine.org/channel/scripting?msg=ZCRd86Xg7vAtDaNKu

Not certain that this is desired, but `CoreStringNames` and `SceneStringNames` aren't used very consistently, several of their `StringName`s are unused, and they're no longer that useful for caching the `StringName`s since we have the `SNAME()` macro. I this will make it less work to implement something better like https://github.com/godotengine/godot-proposals/issues/2152.

### CODE REVIEW

File: core/core_string_names.cpp [hunk 1]
```diff
@@ -1,81 +0,0 @@
...
-
-#include "core_string_names.h"
-
-CoreStringNames *CoreStringNames::singleton = nullptr;
-
-CoreStringNames::CoreStringNames() :
-		_free(StaticCString::create("free")),
-		changed(StaticCString::create("changed")),
-		_meta(StaticCString::create("__meta__")),
-		_script(StaticCString::create("script")),
```
Comment (akien-mga) [inline core/core_string_names.cpp:39]: Note that these ones had stringnames that differ from their identifier, so a simple regex replace will break things.

File: core/variant/variant.cpp [hunk 2]
```diff
@@ -2036,7 +2036,7 @@ Variant::operator ::RID() const {
 		}
 #endif
 		Callable::CallError ce;
-		Variant ret = _get_obj().obj->call(CoreStringNames::get_singleton()->get_rid, nullptr, 0, ce);
+		Variant ret = _get_obj().obj->call(SNAME("get_rid, nullptr, 0, ce");
```
Comment (akien-mga) [inline core/variant/variant.cpp:2039]: Regex didn't handle commas properly.

Comment (LightningAA) [reply core/variant/variant.cpp]: I know, that's why I marked it as draft :). Should be fixed now.

File: core/core_string_names.cpp (hunk 1 above)
Comment (LightningAA) [reply core/core_string_names.cpp:39]: Thanks, I did a quick search to see if any were different, but I missed those.

Comment (LightningAA) [reply core/core_string_names.cpp:39]: I'm not sure if it'd actually break things as long as all occurrences were updated, but better safe than sorry I suppose. Fixed.

**Comment by EricEzaM** [comment]
I think part of the idea was to prevent bugs caused spelling errors in the strings... although the StringNames classes were not really used very diligently or consistently throughout the database. What about `SceneStringNames`?

**Comment by LightningAA** [comment]
> I think part of the idea was to prevent bugs caused spelling errors in the strings... although the StringNames classes were not really used very diligently or consistently throughout the codebase

Didn't know about `SceneStringNames`, if `CoreStringNames` gets removed then it probably should too...
I like https://github.com/godotengine/godot-proposals/issues/2152 better though, I think the signal names should be local to the class they originate from.

> What about `SceneStringNames`, should that be removed too? There may be others.

From a quick search it seems `SceneStringNames` is the only other similar class.

**Comment by LightningAA** [comment]
Alright, I removed `SceneStringNames` too... I was careful to not just do a simple regex replace, but to make sure that the strings actually matched what they were before (I used a regex to check for all StringNames that weren't the same as their variable name).

**Comment by akien-mga** [comment]
I think this may still make sense but would likely need to be redone.

But we should discuss it a bit more beforehand to make sure we agree on what should be done. `SNAME()` can in theory replace these classes but there's a valid point to make that it's not necessarily a gain to rely more on strings (and thus potential typos) instead of caching them in one location.

Back when this was discussed we also had some ideas to maybe use a macro that could define this kind of static StringName in the class with an actual C++ identifier so we avoid strings, but reduz wasn't fond of the idea IIRC.

**Comment by KoBeWi** [comment]
Related: #80573 and #81303
After the latter is merged, I plan to replace all strings with their singleton equivalents if available. The advantage of stringname singletons is that they allocate the name only once, which results in smaller binary size and faster initialization time (see #80573).

While the gain is rather minimal, it's still better to use them in place of repeated SNAMEs and plain strings. Though unused singleton names and names used once should be removed, I plan to clean up that too.

**Comment by akien-mga** [comment]
Superseded by #91909.

**Pull request closed without merging**